### PR TITLE
Implement support for redirects config in the Vercel adapter

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Adds support for redirects in the Vercel adapter by shaping the config.json to the redirect routes.

## Testing

- Test added for static and dynamic routes.

## Docs

Documentation in another issue

### CODE REVIEW

File: packages/integrations/vercel/test/redirects.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	/** @type {import('../../../astro/test/test-utils.js').Fixture} */
+	let fixture;
+
+	before(async () => {
+		fixture = await loadFixture({
+			root: './fixtures/redirects/',
+			redirects: {
+				'/one': '/',
+				'/two': '/',
+				'/blog/[...slug]': '/team/articles/[...slug]',
```
Comment (natemoo-re) [inline packages/integrations/vercel/test/redirects.test.js:15]: What about object syntax?

File: packages/integrations/vercel/src/lib/redirects.ts [hunk 2]
```diff
@@ -54,28 +58,44 @@ function getReplacePattern(segments: RoutePart[][]) {
...
-			if (route.type !== 'page' || route.segments.length === 0) continue;
-
-			redirects.push({
-				src: config.base + getMatchPattern(route.segments) + '/',
-				headers: { Location: config.base + getReplacePattern(route.segments) },
-				status: 308,
-			});
+	for(const route of routes) {
+		if(route.type === 'redirect') {
+			if(true || route.pathname) {
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/integrations/vercel/src/lib/redirects.ts:76]: Yeah, I'll fix this, this code works for dynamic as well.

File: packages/integrations/vercel/test/redirects.test.js (hunk 1 above)
Comment (matthewp) [reply packages/integrations/vercel/test/redirects.test.js:15]: That is in a separate PR (added to the RFC after this was created): https://github.com/withastro/astro/tree/redirects-ssg-object

File: packages/integrations/vercel/src/lib/redirects.ts [hunk 3]
```diff
@@ -36,6 +36,10 @@ function getMatchPattern(segments: RoutePart[][]) {
 		.join('');
 }
 
+function appendTrailingSlash(route: string): string {
+	return route.at(-1) === '/' ? route : route + '/';
+}
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/integrations/vercel/src/lib/redirects.ts]: Ok, it makes me a little nervous, but I went ahead and created a `@astrojs/internal-helpers` and put our path utils there. We can expand in the future.

**Review by natemoo-re — APPROVED** [review]

**Pull request merged**